src/Tempest/View/src/Commands/MakeViewCommand.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\View\Commands;
+
+use InvalidArgumentException;
+use Tempest\Console\ConsoleArgument;
+use Tempest\Console\ConsoleCommand;
+use Tempest\Core\PublishesFiles;
+use Tempest\Generation\DataObjects\StubFile;
+use Tempest\Generation\Enums\StubFileType;
+use Tempest\Generation\Exceptions\FileGenerationAbortedException;
+use Tempest\Generation\Exceptions\FileGenerationFailedException;
+use Tempest\View\Enums\ViewType;
+use Tempest\View\Stubs\ViewStub;
+use function Tempest\Support\str;
+
+final class MakeViewCommand
+{
+    use PublishesFiles;
+
+    #[ConsoleCommand(
+        name: 'make:view',
+        description: 'Creates a view file',
+        aliases: ['view:make', 'view:create', 'create:view'],
+    )]
+    public function __invoke(
+        #[ConsoleArgument(
+            help: 'The file name of the view',
+        )]
+        string $fileName,
+        #[ConsoleArgument(
+            name: 'type',
+            help: 'The type of the view to create',
+        )]
+        ViewType $viewType = ViewType::RAW,
+    ): void {
+        try {
+            $suggestedPath = str($this->getSuggestedPath('Dummy'));
+            $suggestedPath = ($viewType === ViewType::RAW)
+                ? $suggestedPath->replace('Dummy', $fileName . '.view')
+                : $suggestedPath->replace('Dummy', $fileName);
+
+            $suggestedPath = $suggestedPath->toString();
+            $targetPath = $this->promptTargetPath($suggestedPath);
+            $shouldOverride = $this->askForOverride($targetPath);
+
+            $stubFile = $this->getStubFileFromViewType($viewType);
+
+            if ($stubFile->type === StubFileType::RAW_FILE) {
+                $this->stubFileGenerator->generateRawFile(
+                    stubFile: $stubFile,
+                    targetPath: $targetPath,
+                    shouldOverride: $shouldOverride,
+                );
+            } else {
+                $this->stubFileGenerator->generateClassFile(
+                    stubFile: $stubFile,
+                    targetPath: $targetPath,
+                    shouldOverride: $shouldOverride,
+                    replacements: [
+                        'dummy.view.php' => str($fileName)->kebab()->toString() . '.view.php',
+                    ],
+                );
+            }
+
+            $this->success(sprintf('View successfully created at "%s".', $targetPath));
+        } catch (FileGenerationAbortedException|FileGenerationFailedException|InvalidArgumentException $e) {
+            $this->error($e->getMessage());
+        }
+    }
+
+    private function getStubFileFromViewType(ViewType $viewType): StubFile
+    {
+        try {
```

Comment:
I don't think it's necessary to throw an exception type, and re-throw another type after. Can you just return the `match` result?

I'm also curious about that "guardrail", wouldn't phpstan always catch it if we added a new `ViewType`? We don't need to throw here, right?

Reply:
This is because `StubFile` can also throw an `InvalidArgumentException` so we catch this to throw a more specific exception.

For the "guardrail", you're right, phpstan will catch it until we change the config ( I don't think this should happen )
But, it's more a personal preference to have a more specific error message instead of a generic phpstan error.
I can remove it on make commands where I've implemented that if you prefer ?